lib/kubernetes-deploy/statsd.rb

```diff
@@ -22,5 +22,30 @@ def self.build
       end
       ::StatsD.backend
     end
+
+    module MeasureMethods
+      def measure_method(method_name, metric = nil)
+        unless method_defined?(method_name) || private_method_defined?(method_name)
+          raise NotImplementedError, "Cannot instrument undefined method #{method_name}"
+        end
+
+        unless const_defined?("InstrumentationProxy")
+          const_set("InstrumentationProxy", Module.new)
+          should_prepend = true
+        end
+        instrumentation_proxy = const_get("InstrumentationProxy")
```

Comment:
why isn't this just saying `instrumentation_proxy = InstrumentationProxy`?

Reply:
Good question. When I did it that way, the constant seemed to get looked up in the module's namespace instead of the extending class's, whereas `self` when we do `const_set` is the extending class. Error is:`uninitialized constantKubernetesDeploy::StatsD::MeasureMethods::InstrumentationProxy`. I can't say I understand why. `self::InstrumetationProxy` actually works though if that seems better.